test/mocks/BUILD.wip

```diff
@@ -4,6 +4,7 @@ load("//bazel:envoy_build_system.bzl", "envoy_cc_library")
 
 envoy_cc_library(
     name = "common_lib",
+    testonly = 1,
```

Comment:
What is the reason to use `envoy_cc_library` with `testonly = 1` here instead of `envoy_cc_test_library`?